content/java/concepts/hash-map/hash-map.md

```diff
@@ -0,0 +1,217 @@
+---
+Title: 'HashMap'
+Description: 'In Java, a HashMap is used to store items as a "key/value" pairs. The keys and values can be of either same or different types. java // Create a HashMap which stores courses and their teachers as a key/value pair  HashMap<String, String> courseTeacher = new HashMap<String, String>();'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'HashMap'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+In Java, a HashMap is used to store items as key-value pairs. The key and value can be of the same or different data type.
+
+```java
+// Create a HashMap which stores courses and their teachers as a key/value pair
+HashMap<String, String> courseTeacher = new HashMap<String, String>();
```

Comment:
The first code snippet for most Docs entries should be under a "## Syntax" header and be as general as possible. We seem to have such a code snippet already (below this one). 

Can we replace this:

```java
// Create a HashMap which stores courses and their teachers as a key/value pair
HashMap<String, String> courseTeacher = new HashMap<String, String>();
```

with this?

```java
HashMap<KeyDataType, ValueDataType> myHashMap = new HashMap<KeyDataType, ValueDataType>();
```